src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,133 @@
-        if (d == domain::COMPLEX) {
-            return DFTI_COMPLEX;
-        }
-        else {
-            return DFTI_REAL;
-        }
-    }
-
-    constexpr DFTI_CONFIG_VALUE get_precision(precision p) {
-        if (p == precision::SINGLE) {
-            return DFTI_SINGLE;
-        }
-        else {
-            return DFTI_DOUBLE;
-        }
-    }
+    DFT_ERROR status;
+    mklcpu_desc_t device_handle = nullptr;
+    sycl::buffer<mklcpu_desc_t, 1> handle_buffer{ &device_handle, sycl::range<1>{ 1 } };
 
     template <typename... Args>
-    void set_value_item(DFTI_DESCRIPTOR_HANDLE hand, enum DFTI_CONFIG_PARAM name, Args... args) {
-        if (auto ret = DftiSetValue(hand, name, args...); ret != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception(
-                "dft/backends/mklcpu", "set_value_item",
-                "name: " + std::to_string(name) + " error: " + std::to_string(ret));
+    DFT_ERROR set_value_item(mklcpu_desc_t hand, enum DFTI_CONFIG_PARAM name, Args... args) {
+        DFT_ERROR value_err = DFTI_NOTSET;
+        value_err = DftiSetValue(hand, name, args...);
+        if (value_err != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft", "set_value_item", std::to_string(name));
         }
+
+        return value_err;
     }
 
-    void set_value(DFTI_DESCRIPTOR_HANDLE& descHandle,
-                   const detail::dft_values<prec, dom>& config) {
+    void set_value(mklcpu_desc_t& descHandle, dft::detail::dft_values<prec, dom> config) {
         set_value_item(descHandle, DFTI_INPUT_STRIDES, config.input_strides.data());
         set_value_item(descHandle, DFTI_OUTPUT_STRIDES, config.output_strides.data());
         set_value_item(descHandle, DFTI_BACKWARD_SCALE, config.bwd_scale);
         set_value_item(descHandle, DFTI_FORWARD_SCALE, config.fwd_scale);
         set_value_item(descHandle, DFTI_NUMBER_OF_TRANSFORMS, config.number_of_transforms);
         set_value_item(descHandle, DFTI_INPUT_DISTANCE, config.fwd_dist);
         set_value_item(descHandle, DFTI_OUTPUT_DISTANCE, config.bwd_dist);
-        set_value_item(
-            descHandle, DFTI_PLACEMENT,
-            (config.placement == config_value::INPLACE) ? DFTI_INPLACE : DFTI_NOT_INPLACE);
+        set_value_item(descHandle, DFTI_COMPLEX_STORAGE,
+                       to_mklcpu<config_param::COMPLEX_STORAGE>(config.complex_storage));
+        set_value_item(descHandle, DFTI_REAL_STORAGE,
+                       to_mklcpu<config_param::REAL_STORAGE>(config.real_storage));
+        set_value_item(descHandle, DFTI_CONJUGATE_EVEN_STORAGE,
+                       to_mklcpu<config_param::CONJUGATE_EVEN_STORAGE>(config.conj_even_storage));
+        set_value_item(descHandle, DFTI_PLACEMENT,
+                       to_mklcpu<config_param::PLACEMENT>(config.placement));
+        set_value_item(descHandle, DFTI_PACKED_FORMAT,
+                       to_mklcpu<config_param::PACKED_FORMAT>(config.packed_format));
```

Comment:
* FORWARD_DOMAIN  ✔️ (in descriptor creation)
* DIMENSION  ✔️ (in descriptor creation)
* LENGTHS ✔️ (in descriptor creation)
* PRECISION ✔️ (in descriptor creation)
* FORWARD_SCALE, ✔️ 
* BACKWARD_SCALE ✔️ 
* NUMBER_OF_TRANSFORMS ✔️ 
* COMPLEX_STORAGE ✔️ 
* REAL_STORAGE ✔️ 
* CONJUGATE_EVEN_STORAGE ✔️ 
* PLACEMENT ✔️ 
* INPUT_STRIDES  ✔️ 
* OUTPUT_STRIDES  ✔️ 
* FWD_DISTANCE ✔️ 
* BWD_DISTANCE ✔️ 
* WORKSPACE 
* ORDERING 
* TRANSPOSE 
* PACKED_FORMAT ✔️ 
Looks like some config values are missing compared to the [specification](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/dft/enums_and_config_params.html#onemkl-dft-enum-config-param) I'm looking at.

Reply:
@hjabird fixed in [2e14329](https://github.com/oneapi-src/oneMKL/pull/288/commits/2e14329e390d73b6a93e9d216279fd05e31ee45f), please check

Reply:
All above items now handled in some way. Please see new comment https://github.com/oneapi-src/oneMKL/pull/288/commits/2e14329e390d73b6a93e9d216279fd05e31ee45f#r1166620376

Reply:
addressed in b56bf891cee45fc2381148f854ac7d4215cfb0c0, resolving as minor correction.